core/chain_makers.go

```diff
@@ -164,7 +165,7 @@ func (b *BlockGen) OffsetTime(seconds int64) {
 // values. Inserting them into BlockChain requires use of FakePow or
 // a similar non-validating proof of work implementation.
 func GenerateChain(parent *types.Block, db ethdb.Database, n int, gen func(int, *BlockGen)) ([]*types.Block, []types.Receipts) {
-	statedb, err := state.New(parent.Root(), db)
+	statedb, err := state.New(parent.Root(), access.NewDbChainAccess(db))
```

Comment:
Don't create a new access layer every time you want to generate a chain. Pass it as a parameter instead of the `db ethdb.Database`. Maybe @fjl has an input on this? But IMHO if we replace the databases with accessor objects, then they should be fully replaced from their point of creation and not just wrapped at the last minute (would be very very wasteful memory wise to have a lot of small wraps deep in the code).